pkg/appruntime/documentloader/documentloader.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2024 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package documentloader
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	documentloaders "github.com/tmc/langchaingo/documentloaders"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/textsplitter"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubeagi/arcadia/api/app-node/documentloader/v1alpha1"
+	arcadiav1alpha1 "github.com/kubeagi/arcadia/api/base/v1alpha1"
+	"github.com/kubeagi/arcadia/pkg/appruntime/base"
+	"github.com/kubeagi/arcadia/pkg/config"
+	"github.com/kubeagi/arcadia/pkg/datasource"
+	arcadiadocumentloaders "github.com/kubeagi/arcadia/pkg/documentloaders"
+)
+
+type Executor struct {
+	base.BaseNode
+}
+
+func NewExecutor(baseNode base.BaseNode) *Executor {
+	return &Executor{
+		baseNode,
+	}
+}
+
+func (p *Executor) Run(ctx context.Context, cli client.Client, args map[string]any) (map[string]any, error) {
+	instance := &v1alpha1.DocumentLoader{}
+	// Check if have docs as input
+	v1, ok := args["files"]
+	if !ok {
+		return args, errors.New("no input docs")
+	}
+	files, ok := v1.([]string)
+	if !ok || len(files) == 0 {
+		return args, errors.New("empty file list")
+	}
+	if err := cli.Get(ctx, types.NamespacedName{Namespace: p.RefNamespace(), Name: p.Ref.Name}, instance); err != nil {
+		return args, fmt.Errorf("can't find the documentloader in cluster: %w", err)
+	}
+
+	system, err := config.GetSystemDatasource(ctx, cli)
+	if err != nil {
+		return nil, err
+	}
+	endpoint := system.Spec.Endpoint.DeepCopy()
+	if endpoint != nil && endpoint.AuthSecret != nil {
+		endpoint.AuthSecret.WithNameSpace(system.Namespace)
+	}
+	ossDatasource, err := datasource.NewLocal(ctx, cli, endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var allDocs []schema.Document
+	var textArray []string
+
+	for _, file := range files {
+		ossInfo := &arcadiav1alpha1.OSS{Bucket: p.RefNamespace()}
+		ossInfo.Object = filepath.Join("upload", file)
+		klog.Infoln("handling file", ossInfo.Object)
```

Comment:
we should use `klog.FromContext(ctx).xxx` to replace `klog`, We add a `requestid` to each request in the logs to make debugging easier, in
https://github.com/kubeagi/arcadia/blob/352145b8ab110e4d8abdde4412c9dcc5d94bfece/apiserver/pkg/requestid/requestid.go#L29-L36

But it looks like this is an issue elsewhere as well, and we can leave that for a subsequent pr along with an update.

Reply:
yes, klog it more easy to use